local-cli/init/init.js

```diff
@@ -8,48 +8,104 @@
  * Creates the template for a React Native project given the provided
  * parameters:
- *   - projectDir: templates will be copied here.
- *   - argsOrName: project name or full list of custom arguments to pass to the
- *                 generator.
+ * @param projectDir Templates will be copied here.
+ * @param argsOrName Project name or full list of custom arguments
+ *                   for the generator.
  */
 function init(projectDir, argsOrName) {
   console.log('Setting up new React Native app in ' + projectDir);
-  const env = yeoman.createEnv(
-    undefined,
-    undefined,
-    new CreateSuppressingTerminalAdapter()
-  );
 
-  env.register(
-    require.resolve(path.join(__dirname, '../generator')),
-    'react:app'
+  const args = Array.isArray(argsOrName)
+    ? argsOrName // ['AwesomeApp', '--verbose']
+    : [argsOrName].concat(process.argv.slice(4)); // 'AwesomeApp'
+
+  // args is ['AwesomeApp', '--verbose']
+  if (!args || args.lentgh == 0) {
+    console.error('react-native init requires a project name.');
+    return;
+  }
+
+  const newProjectName = args[0];
+  const options = minimist(args);
+
+  generateProject(projectDir, newProjectName, options);
+}
+
+/**
+ * Generates a new React Native project based on the template.
+ * @param Absolute path at which the project folder should be created.
+ * @param options Command line arguments parsed by minimist.
+ */
+function generateProject(destinationRoot, newProjectName, options) {
+  var reactNativePackageJson = require('../../package.json');
+  var { peerDependencies } = reactNativePackageJson;
+  if (!peerDependencies) {
+    console.error('Missing React peer dependency in React Native\'s package.json. Aborting.');
+    return;
+  }
+
+  var reactVersion = peerDependencies.react;
+  if (!reactVersion) {
+    console.error('Missing React peer dependency in React Native\'s package.json. Aborting.');
+    return;
+  }
+
+  const yarnVersion =
+    (!options['npm']) &&
+    yarn.getYarnVersionIfAvailable() &&
+    yarn.isGlobalCliUsingYarn(destinationRoot);
+
+  copyProjectTemplateAndReplace(
+    path.resolve('node_modules', 'react-native', 'local-cli', 'templates', 'HelloWorld'),
+    destinationRoot,
+    newProjectName
   );
 
-  // argv is for instance
-  // ['node', 'react-native', 'init', 'AwesomeApp', '--verbose']
-  // args should be ['AwesomeApp', '--verbose']
-  const args = Array.isArray(argsOrName)
-    ? argsOrName
-    : [argsOrName].concat(process.argv.slice(4));
+  if (yarnVersion) {
+    console.log('Adding React...');
+    execSync(`yarn add react@${reactVersion}`);
+  } else {
+    console.log('Installing React...');
+    execSync(`npm install react@${reactVersion} --save --save-exact`);
+  }
+  if (!options['skip-jest']) {
+    const jestDeps = `jest babel-jest jest-react-native babel-preset-react-native react-test-renderer@${reactVersion}`;
+    if (yarnVersion) {
+      console.log('Adding Jest...');
+      execSync(`yarn add ${jestDeps} --dev --exact`);
```

Comment:
Martin, while you are at it, how about we provide a backdoor for CI systems and offline installs.
Recently Legocastle was not able to block a bug because we can't reproduce install offline.

Any ideas?
From the top of my head we could provide an override CLI command of some sort here.
And compose this command on CI to fake yarn install